server/coordinator.go

```diff
@@ -109,6 +109,21 @@ func (c *coordinator) dispatch(region *core.RegionInfo) {
 			c.removeOperator(op)
 		}
 	}
+	// If PD has restarted, it need to check learners added before and promote them.
+	if c.cluster.IsEnableRaftLearner() && c.getOperator(region.GetId()) == nil {
```

Comment:
Maybe better to put it into `replicaChecker`?

Reply:
It needs coordinator itself to call `getOperator()` but replicaChecker has no way to it.